daemon/daemon.go

```diff
@@ -8,26 +8,193 @@ import (
 	"time"
 
 	"github.com/CanonicalLtd/UCWifiConnect/netman"
+	"github.com/CanonicalLtd/UCWifiConnect/server"
+	"github.com/CanonicalLtd/UCWifiConnect/utils"
+	"github.com/CanonicalLtd/UCWifiConnect/wifiap"
 )
 
+// enum to track current system state
+const (
+	unknown = iota
+	managed
+	operational
+)
+
+//used to clase the operataional http server
+var err error
+
+// scanSsids sets wlan0 to be managed and then scans
+// for ssids. If found, write the ssids (comma separated)
+// to path and return true, else return false.
+func scanSsids(path string, c *netman.Client) bool {
+	manage(c)
+	time.Sleep(5000 * time.Millisecond)
+	SSIDs, _, _ := c.Ssids()
+	//only write SSIDs when found
+	if len(SSIDs) > 0 {
+		var out string
+		for _, ssid := range SSIDs {
+			out += strings.TrimSpace(ssid.Ssid) + ","
+		}
+		out = out[:len(out)-1]
+		err := ioutil.WriteFile(path, []byte(out), 0644)
+		if err != nil {
+			fmt.Println("==== Error writing SSID(s) to ", path)
+		} else {
+			fmt.Println("==== SSID(s) found and written to ", path)
+			return true
+		}
+	}
+	fmt.Println("==== NO SSID found")
+	return false
+}
+
+// unmanage sets wlan0 to be unmanaged by network manager if it
+// is managed
+func unmanage(c *netman.Client) {
+	ifaces, _ := c.WifisManaged(c.GetWifiDevices(c.GetDevices()))
+	if _, ok := ifaces["wlan0"]; ok {
```

Comment:
valid for V1, but take care of wlan0 could be another interface name in future. Maybe would be useful holding this in a const